tinygrad/nn/__init__.py

```diff
@@ -120,5 +120,5 @@ def __init__(self, vocab_size:int, embed_size:int):
     self.weight = Tensor.glorot_uniform(vocab_size, embed_size)
 
   def __call__(self, idx:Tensor) -> Tensor:
-    vocab_counter = Tensor.arange(self.vocab_size, requires_grad=False).reshape(1, 1, self.vocab_size).expand(*idx.shape, self.vocab_size)
-    return (vocab_counter == idx.unsqueeze(2).expand(*idx.shape, self.vocab_size)) @ self.weight
+    if not hasattr(self, 'vocab_counter'): self.vocab_counter = Tensor.arange(self.vocab_size, requires_grad=False).reshape(1, 1, self.vocab_size)
```

Comment:
is there a benefit of having this in `__call__` rather than in `__init__`? cuz we already know the `vocab_size` in during initialization.

Reply:
The issue with putting it in init is that it is part of the state and the load fails

Reply:
ah, hmm. would having something in nn/state.py that ignores certain attributes help?